v3/src/components/case-table/case-table.tsx

```diff
@@ -37,16 +35,26 @@ export const CaseTable = observer(function CaseTable({ setNodeRef }: IProps) {
     setNodeRef(elt)
   }
 
-  useEffect(function syncScrollLeft() {
-    // There is a bug, seemingly in React, in which the scrollLeft property gets reset
-    // to 0 when the order of tiles is changed (which happens on selecting/focusing tiles
-    // in the free tile layout), even though the CaseTable component is not re-rendered
-    // or unmounted/mounted. Therefore, we reset the scrollLeft property from our saved
-    // cache on focus change.
-    if (isFocused && contentRef.current) {
-      contentRef.current.scrollLeft = tableModel?.scrollLeft ?? 0
+  useEffect(() => {
+    const updateScroll = (horizontalScrollOffset?: number) => {
+      if (horizontalScrollOffset != null && contentRef.current &&
+        contentRef.current.scrollLeft !== horizontalScrollOffset
+      ) {
+        contentRef.current.scrollLeft = horizontalScrollOffset
+      }
     }
-  }, [isFocused, tableModel])
+
+    // Initial scoll is delayed a frame to let RDG do its thing
+    setTimeout(() => updateScroll(tableModel?.horizontalScrollOffset))
+
+    // Handle changes to the model, such as via the API
+    return reaction(
+      () => tableModel?.horizontalScrollOffset,
+      horizontalScrollOffset => {
+        updateScroll(horizontalScrollOffset)
+      }
+    )
```

Comment:
Should use `mstReaction` here so that the reaction is disposed when the `tableModel` is destroyed. Also, if you pass `fireImmediately: true`, you could avoid the separate call to `updateScroll` on initialization, although you'd presumably have to use `setTimeout` inside the reaction in that case, so not clear whether that's a win or not. 🤷

Reply:
I was originally using `fireImmediately: true`, but switched it so the `setTimeout` is only for the initial load. My preference is to minimize `setTimeout` (I'm not even happy about the one!) but if you prefer the other way around I'll switch it.

Reply:
No, I think you're right that restricting the use of `setTimeout` to the situations we need it is probably a higher priority. I just wanted to point out the option so the options was considered. There is such thing as being too DRY.